django/gregory/functions.py

```diff
@@ -24,3 +36,121 @@ def get_doi(title):
+		dataset["summary"] = dataset["summary"].apply(html.unescape)
+
+		# Now let's remove all those HTML tags
+		dataset["summary"] = dataset["summary"].apply(cleanHTML)
+
+		# Now let's clean the resulting text in the summary column
+		dataset["summary"] = dataset["summary"].apply(cleanText)
+
+		# Let's join the two text columns into a single 'terms' column
+		dataset["terms"] = dataset["title"].astype(str) + " " + dataset["summary"].astype(str)
+
+		# We no longer need some text columns, let's remove them
+		dataset = dataset[["terms", "relevant", "article_id"]]
+
+		# There are several records in the "relevant" column as NaN. Let's convert them to zeros
+		dataset["relevant"] = dataset["relevant"].fillna(value=0)
+
+		# A value is trying to be set on a copy of a slice from a DataFrame.
+		# Try using .loc[row_indexer,col_indexer] = value instead
+
+		# See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
+
+		# change true/false to 1/0
+		dataset["relevant"] = dataset["relevant"].astype(int)
+
+		# Save the dataset
+		dataset.to_csv(dataset_file_csv, index=False)
+
+		# Load the source data into a Pandas dataframe
+		dataset = pd.read_csv(dataset_file_csv)
+
+		# Replace any NaN with zero
+		dataset['relevant'] = dataset['relevant'].fillna(value=0)
+
+		# Models to use from the list above
+		models = [GNB,LR]
+
+		# This is the dict that will store the pipelines
+		pipelines = {}
+
+		for model in models:
+			pipelines[model] = load('/code/gregory/ml_models/model_' + model + '.joblib')
+
+		def predictor(dataset):
+			data = {"O": []}
+			result = {}
+			result["models"] = {}
+			for model in models:
+				data[model] = []    
+				result["models"][model] = []
+			for index, row in dataset.iterrows():
+				input = row['terms']
+				output = row['relevant']
+				data["O"].append(output)
+				test = "TEST " + str(index + 1) + " (provided: " + str(int(output)) + ")"
+				for model in models:
+					prediction = pipelines[model].predict([input])
+					data[model].append(prediction)
+					test += " - " + model + ": " + str(prediction)
+					result["models"][model].append({
+						"article_id": row['article_id'],
+						"prediction": str(prediction)
+					})
+			return result,data
+
+		data = predictor(dataset)
+		for model in data[0]['models']:
+			for item in data[0]['models'][model]:
+				# print(model)
+				article = Articles.objects.get(pk=item['article_id'])
+				# print(article,model,item['prediction'],type(item['prediction']))
+				if item['prediction'] == '1' or item['prediction'] == "['True']":
+					if model == 'gnb':
+						article.ml_prediction_gnb = True
+					if model == 'lr':
+						article.ml_prediction_lr = True
+				if item['prediction'] == "['0']":
+					if model == 'gnb':
+						article.ml_prediction_gnb = False
+					if model == 'lr':
+						article.ml_prediction_lr = False
+				article.save()
```

Comment:
Did you debug the application to make sure it actually reaches this line where it commits the data to the database?

Reply:
Did that this morning. The problem is that I was passing a django object with `Articles.objects.get()` and it wasn't being converted to a pandas dataframe.

Changing to `Articles.objects.filter()` resolved the issue.